src/crypto/algorithms/megolm.ts

```diff
@@ -265,6 +265,21 @@ class MegolmEncryption extends EncryptionAlgorithm {
      * @param {boolean} [singleOlmCreationPhase] Only perform one round of olm
      *     session creation
      *
+     * This method updates the setupPromise field of the class by chaining a new
+     * call on top of the existing promise, and then catching and discarding any
+     * errors that might happen while setting up the outbound group session. This
+     * is done to ensure that `setupPromise` always resolves to `null` or the
+     * `OutboundSessionInfo`.
+     *
+     * Using `>>=` to represent the promise chaining operation, it does the
+     * following:
+     *
+     * setupPromise = previousSetupPromise >>= setup >>= discardErrors
+     *
+     * The initial value for the `setupPromise` is a promise that resolves to
+     * `null`. The forceDiscardSession() resets setupPromise to this initial
+     * promise.
```

Comment:
it seems like this is an implementation detail which belongs in a comment inside the function, rather than in the doc-comment?

Reply:
Considering that this method is private, and modifies a field of the class, I would say that it's perfectly reasonable to explain how it does so in the doc comment.

Reply:
Not entirely convinced, but I don't feel that strongly.